### PR TITLE
Enhancement: 6-7 Add GET Endpoints

### LINKED ISSUE #6: GET Challenge
As a user of the REST endpoint I should get a response of a specific challenge from the endpoint: `/v1/challenge/:id`.

URL: `/v1/challenge/:id`

## Response

```js
{
  "id": String,
  "level": String,
  "question": String,
  "source": String
}
```

## Response example

```json
{
  "id": "1",
  "level": "1",
  "question": "Write a program which will find all such numbers which are divisible by 7 but are not a multiple of 5, between 2000 and 3200 (both included). The numbers obtained should be printed in a comma-separated sequence on a single line.",
  "source": "none"
}
```

### LINKED ISSUE #7: GET Random Challenge
As a user of the REST endpoint I should get a response of a random challenge from the endpoint: `/v1/challenge`.

URL: `/v1/challenge/`

## Response

```js
{
  "id": String,
  "level": String,
  "question": String,
  "source": String
}
```

## Response example

```json
{
  "id": "1",
  "level": "1",
  "question": "Write a program which will find all such numbers which are divisible by 7 but are not a multiple of 5, between 2000 and 3200 (both included). The numbers obtained should be printed in a comma-separated sequence on a single line.",
  "source": "none"
}
```

### PR DESCRIPTION
**Pull request by bmw2621**
ISSUE:
Enhancement
Issue 6: Add Challenge Endpoint
Issue 7: Add Random Challenge Enpoint
STEPS TO TEST:
Tested with Postman

## Description

Added endpoints for getting a random challenge and getting a challenge by id

## Related Issue

CLOSES #6 
CLOSES #7 

## How Has This Been Tested?

Tested in Postman - Pending setup of testing framework

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by TheSavageDev** [comment]
Link issues using `KEYWORD #6`

**Comment by TheSavageDev** [comment]
You can do multiple `Closes #6, Closes #7`

File: src/routes/v1/challenge.route.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+router
+  .route("/")
+  .get((req, res) => challengeController.getRandomChallenge(req, res));
+
+router
+  .route("/:id")
+  .get((req, res) => challengeController.getChallengeById(req, res));
+
+module.exports = router;
```
Comment (TheSavageDev) [inline src/routes/v1/challenge.route.js:13]: Swagger?

**Review by TheSavageDev — CHANGES_REQUESTED** [review]

File: src/routes/v1/challenge.route.js (hunk 1 above)
Comment (bmw2621) [reply src/routes/v1/challenge.route.js:13]: I was going to do it in a different PR.  But I can do it on this one

**Review by TheSavageDev — APPROVED** [review]

**Pull request merged**